packages/cli/src/ceramic-daemon.ts

```diff
@@ -284,6 +285,17 @@ export class CeramicDaemon {
     )
     const ipfsId = await ipfs.id()
 
+    // we need to change the values before processing the config, so we keep track to log the warning after we have logging set up
+    const desiredIpfsClient = EnvironmentUtils.useRustCeramic() ? 'ceramic-one' : 'kubo'
+    const wrongIpfsClient =
+      opts.ipfs.mode == IpfsMode.REMOTE &&
+      EnvironmentUtils.useRustCeramic() &&
+      !ipfsId.agentVersion.includes('ceramic-one')
+
+    if (wrongIpfsClient) {
+      EnvironmentUtils.setIpfsFlavor(EnvironmentUtils.useRustCeramic() ? 'go' : 'rust')
```

Comment:
We definitely want to toggle the flavor instead of erroring out? I'd be worried about an inconsistency in the config resulting in unexpected behavior.

In fact, if someone were to upgrade to this version of the code on an existing remote setup using Kubo, Ceramic will try to communicate with Kubo as if it is C1 because they might not have set `IPFS_FLAVOR=go` explicitly.

Reply:
I don't think that's true.. but I could very well be wrong 😅. As of #3250, IPFS_FLAVOR is rust by default. If we specify a remote with `--ipfs-url` in this branch, we'll detect that rust != kubo, log a warning and use kubo IPFS.

Reply:
This is targeting #3250 currently rather than develop.

Reply:
It seems like there should be a distinction between if you're using the default behavior vs if you set IPFS_FLAVOR explicitly.  If the env var is set explicitly and differs from what is detected, that feels like it should fail startup.  Whereas if it's unset and you're using the defaults, then adjusting to whatever is detected makes sense.

Reply:
Improved the warnings. If you specify `IPFS_FLAVOR=go` with a kubo `--ipfs-url` there is no warning, otherwise there is. We log a warning if you specified flavor=go and talked to c1 with a url, but we can just "make it work" so I'm not sure there's a need to crash. We can change the warning to an error though.